services/galley/src/Galley/API/Internal.hs

```diff
@@ -55,17 +56,17 @@ rmUserH (user ::: conn) = do
 rmUser :: UserId -> Maybe ConnId -> Galley ()
 rmUser user conn = do
   let n = unsafeRange 100 :: Range 1 100 Int32
-  Data.ResultSet tids <- Data.teamIdsFrom user Nothing (rcast n)
+  tids <- Data.teamIdsForPagination user Nothing (rcast n)
   leaveTeams tids
-  Data.ResultSet cids <- Data.conversationIdsFrom user Nothing (rcast n)
+  cids <- Data.conversationIdsForPagination user Nothing (rcast n)
```

Comment:
Pagination with `Data.Result` was broken due to the usage of `+1` when fetching the size.

Reply:
The `ResultSet` newtype type exists to show that `nextPage` cannot be used reliably. Still, it was just unwrapped and `nextPage` was used.

Maybe instead of just wrapping `Page`, `ResultSet` should contain only the `hasMore` and `result` fields? I'll put that on my list, we might find more bugs.

Reply:
I had already commented here... did Github lose my comment? Note that the constructor is no longer exported so that's already taken care of.